Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,41 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+
+        TensorDimensions outputSizes = {};
+        uint32_t totalElements = 1;
+        for (uint32_t i = 0; i < dimensionCount; ++i)
+        {
+            totalElements *= inputTensorSizes[i];
+            outputSizes.push_back(1);
+        }
+        TensorDesc outputCountTensor(DML_TENSOR_DATA_TYPE_UINT32, outputSizes, builder->GetTensorPolicy());
+        TensorDesc outputCoordinatesTensor(DML_TENSOR_DATA_TYPE_UINT32, {totalElements, dimensionCount}, builder->GetTensorPolicy());
+
+        DML_NONZERO_COORDINATES_OPERATOR_DESC desc = {};
+        desc.InputTensor = inputTensor.AsPtr<DML_TENSOR_DESC>();
+        desc.OutputCountTensor = outputCountTensor.AsPtr<DML_TENSOR_DESC>();
+        desc.OutputCoordinatesTensor = outputCoordinatesTensor.AsPtr<DML_TENSOR_DESC>();
+        
+        NonZeroCoordinatesOutputs out;
```

Comment:
>out [](http://example.com/codeflow?start=34&length=3)

[nit] `output`, since you used `output` everywhere else (e.g. it's not outCountTensor, outSizes...).